packages/contracts/package.json

```
@@ -33,7 +33,7 @@
     "clean": "run-p clean:*",
     "clean:yarn": "yarn clean",
     "clean:forge": "forge clean",
-    "build": "forge build",
+    "build": "tsx ./scripts/prebuild/export-smtchecker-contracts.ts && forge build",
```

```suggestion
    "build": "run-s build:*",
    "build:smt": "tsx ./scripts/prebuild/export-smtchecker-contracts.ts",
    "build:compile": "forge build",
```

`run-s` should run those in series. You can also parallelize with `run-p`

I didn't actually test this here, I only worked on it in this comment, so you might want to verify I have no syntax errors etc before moving on.

---

You can also specify the order via `run-s build:smt build:compile`

---

You don't have the `npm-run-all` dependency on that workspace, do you want me to add it? Apart from that, you don't have any tasks, you can only run `run-p build` or `run-s build`

---

> don't have the `npm-run-all` dependency on that workspace

That's strange I didn't notice that before. Yeah I think adding it is fine!

> you don't have any tasks

My code suggestion is to add those two sub tasks so you can combine them with `run-s build:smt build:compile`

---

Since SMTChecker is included in the foundry config file, I can't find how to ignore the SMTChecker when running `forge build` (in order to separate the smt from the compile).

I thought about 'over-engineering' it and run the script to add/remove the SMTChecker config from the `foundry.toml`. But it seems like a bad solution.

Do you have any ideas?

---

I have basically zero formal Foundry experience so it's best if we get input from some of the team. @rndquu @zgorizzo69 @0xcodercrane

---

what about creating a smt profile https://book.getfoundry.sh/config/?highlight=profile#configuring-with-foundrytoml so i.e 
if `FOUNDRY_PROFILE` is smtchecker then will run smt checker when running forge build otherwise it will not

---

> what about creating a smt profile https://book.getfoundry.sh/config/?highlight=profile#configuring-with-foundrytoml so i.e if `FOUNDRY_PROFILE` is smtchecker then will run smt checker when running forge build otherwise it will not

Good point. @jpldcarvalho you could create 2 profiles: default (without smt checker) and the one which enables smt. Then run `forge build` using the smt profile.